src/controller.rs

```diff
@@ -349,10 +349,21 @@ pub async fn run_cluster_controller(state: State) {
         .default_backoff()
         .for_each(|_| futures::future::ready(()));
 
+    let cluster_ns_controller = Controller::new(
+        Api::<Cluster>::all(client.clone()),Default::default())
+        .shutdown_on_signal()
+        .run(
+            Cluster::reconcile_fleet_annotation_in_capi_ns,
+            error_policy,
+            state.to_context(client.clone()),
+        )
+        .default_backoff()
+        .for_each(|_| futures::future::ready(()));
+
```

Comment:
I think it would be better to reuse cluster cache, which is narrowed down to label selectors, and follow the pattern of

https://github.com/rancher/cluster-api-addon-provider-fleet/blob/5aaff1b8b9b9ee57faa436401377470c5ef04088/src/controller.rs#L329-L350

Here it creates a fully separate cluster-wide CAPI cluster cache.